### DISCUSSION TITLE
Open nvdash when delete all buffer action and last empty buffer [no name]

### DISCUSSION THREAD

**Discussion by asapdotid** [NvChad/NvChad#2896] [Q&A] [answered]
Hello!

I'd like for `nvdash` to load up whenever I close all the open, non-tree buffers.

Currently, If i close all those buffers, I get an empty [No Name] buffer, I'd like to see `nvdash` instead.

How? tq

**Comment by siduck** [comment]
@asapdotid you've to create an autocmd on evens like bufadd , bufDelete . 

use    our `vm.t.bufs` variable, it stores buffer numbers, and just check if it has 1 item + the bufnr's  bufname is ""

**Reply by asapdotid** [reply]
hi @siduck how to hide or remove tab bar on load nvdash.
![2024-06-19_11-38](https://github.com/NvChad/NvChad/assets/34257858/e7c352a7-24c6-43ad-8a07-766d5c8fb0bd)

**Reply by siduck** [reply]
@asapdotid  showtabline opt

**Reply by asapdotid** [reply]
@siduck I'm a new Vim & Lua user, I'm trying to make this code.

```lua
local nvdash_start_group = vim.api.nvim_create_augroup("NvdashStart", { clear = true })

vim.api.nvim_create_autocmd({ "BufEnter", "BufLeave" }, {
  pattern = "*",
  group = nvdash_start_group,
  callback = function()
    if (vim.tbl_contains({ "" }, vim.o.filetype)) and (#vim.api.nvim_list_wins() == 1) then
      vim.schedule(function()
        local buf_lines = vim.api.nvim_buf_get_lines(0, 0, 1, false)
        local no_buf_content = vim.api.nvim_buf_line_count(0) == 1 and buf_lines[1] == ""
        local bufname = vim.api.nvim_buf_get_name(0)

        if bufname == "" and no_buf_content then
          require("nvchad.nvdash").open()
          require("nvchad.utils").set_cleanbuf_opts "nvdash"
        end
      end, 0)
    end
  end,
})

local type_group = "_dashboard_settings"
vim.api.nvim_create_augroup(type_group, {})
vim.api.nvim_create_autocmd("FileType", {
  group = type_group,
  pattern = "*",
  callback = function()
    vim.schedule(function()
      if vim.bo.ft == "nvdash" then
        vim.opt.showtabline = 0
      else
        vim.opt.showtabline = 2
      end
    end, 0)
  end,
})
```

What's your thought?

**Reply by siduck** [reply, marked as answer]
@asapdotid  your method is alright but no need of vim.schedule in 2nd autocmd, and idk why we need the augroup or pattern

```lua
vim.api.nvim_create_autocmd({ "BufEnter", "BufDelete", "FileType" }, {
  callback = function(args)
    if args.event == "FileType" then
      vim.o.showtabline = vim.bo.ft == "nvdash" and 0 or 2
      return
    end

    local buf = args.buf

    if not vim.bo[buf].buflisted then
      return
    end

    vim.schedule(function()
      if #vim.t.bufs == 1 and vim.api.nvim_buf_get_name(buf) == "" then
        vim.cmd "Nvdash"
      end
    end)
  end,
})
```

**Reply by asapdotid** [reply]
@siduck I have tried with your code it works, thanks for your help.

**Reply by bj5ops** [reply]
> @asapdotid your method is alright but no need of vim.schedule in 2nd autocmd, and idk why we need the augroup or pattern
> 
> ```lua
> vim.api.nvim_create_autocmd({ "BufEnter", "BufDelete", "FileType" }, {
>   callback = function(args)
>     if args.event == "FileType" then
>       vim.o.showtabline = vim.bo.ft == "nvdash" and 0 or 2
>       return
>     end
> 
>     local buf = args.buf
> 
>     if not vim.bo[buf].buflisted then
>       return
>     end
> 
>     vim.schedule(function()
>       if #vim.t.bufs == 1 and vim.api.nvim_buf_get_name(buf) == "" then
>         vim.cmd "Nvdash"
>       end
>     end)
>   end,
> })
> ```

@siduck how for the code to adjust to the new settings NVCHAD UI

**Reply by siduck** [reply]
@bj5ops  it shouldnt matter, i just refactored nvdash

**Reply by bj5ops** [reply]
@siduck on Telescope found `Cannot make changes, 'modifiable' is off`

**Reply by siduck** [reply]
how's telescope related to this @bj5ops , seek help in our discord server

**Reply by asapdotid** [reply]
@siduck with new ui, if using your code, make an impact ontelescope input

**Reply by siduck** [reply]
@asapdotid i dont understand, make a video
